src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -419,13 +425,21 @@ public ProgrammingExercise updateProgrammingExercise(ProgrammingExercise program
         setURLsForAuxiliaryRepositoriesOfExercise(updatedProgrammingExercise);
         connectAuxiliaryRepositoriesToExercise(updatedProgrammingExercise);
 
+        // Make sure that the original references are preserved and the channel is updated if necessary
+        Channel channel = channelService.updateExerciseChannel(programmingExerciseBeforeUpdate, updatedProgrammingExercise);
+
         ProgrammingExercise savedProgrammingExercise = programmingExerciseRepository.save(updatedProgrammingExercise);
 
         participationRepository.removeIndividualDueDatesIfBeforeDueDate(savedProgrammingExercise, programmingExerciseBeforeUpdate.getDueDate());
         programmingExerciseTaskService.updateTasksFromProblemStatement(savedProgrammingExercise);
         // TODO: in case of an exam exercise, this is not necessary
         scheduleOperations(updatedProgrammingExercise.getId());
         groupNotificationScheduleService.checkAndCreateAppropriateNotificationsWhenUpdatingExercise(programmingExerciseBeforeUpdate, savedProgrammingExercise, notificationText);
+
+        if (channel != null) {
+            savedProgrammingExercise.setChannelName(channel.getName());
```

Comment:
Why is this needed? The channel got already update above and the value won't get saved.

Reply:
Otherwise, the savedProgrammingExercise object does not include the channelName. We have to check channel for null, since it is null for exam exercises.

Reply:
Yes, but why does this value need to be set later on? Must it be sent to the client? Because at least on the server I don't see it used.